sql/core/src/test/scala/org/apache/spark/sql/CTEInlineSuite.scala

```diff
@@ -735,6 +735,27 @@ abstract class CTEInlineSuiteBase
       checkAnswer(df, Row(1))
     }
   }
+
+  test("SPARK-49816: Conflicting CTE ids 2") {
```

Comment:
I wanted to add this test because of https://github.com/apache/spark/pull/48284#issuecomment-2391932052.

But actually this test revealed a new, unrelated issue that random seeds are not regenerated when a view is inserted into a plan.